examples/TEM_data/README.md

```diff
@@ -0,0 +1,381 @@
+  statistics.  *IRI:* [pm:precipitation_model_input].
+
+**Transformations**
+- **Image analyser**. This resource is a (hypothetical) *TEM image analyser*
+  software that function both as a sink (takes a DM3 file as input)
+  and a source (calculates precipitation statistics).  It is implemented as a
+  Python function. *IRI:* [pm:image_analyser].
+
+
+Workflow
+--------
+The overall workflow for this example is shown in the figure below:
+
+![resources](figs/resources.svg)
+
+*Figure: Schematics of the overall workflow for this example.
+After the documentation is done, the user can mix and match data
+sources and sinks in a very flexible way.*
+
+All the data resources are associated with a data model (DLite Metadata).
+The microscope settings is associated with the [TEMImage] datamodel and
+the TEM image analyser takes a [TEMImage] instance as input and returns a
+[PrecipitateStatistics] instance.
+
+The workflow includes three different ways to connect a data sink to a
+data source:
+
+* **Generate**. In this case the sink and source shares the same data
+  model.  The generate step simply serialises the source data model in
+  the desired format. A generation step is normally also applied after
+  property mappings and conversion.  The thumbnail image (png),
+  microscope settings (json), precipitate statistics (csv) and the
+  precipitation model input (txt) are all produced by a generator.
+
+* **Property mappings** allows to automatically create and populate an
+  instance of a data model that represents the data as required by the
+  user.  See the [mappings example] for more information.  Property
+  mappings are here used to both instantiate an [Image] data model
+  from a [TEMImage] instance and to instantiate a [Chemistry] by
+  combining a [PrecipitateStatistics] and [Composition] data model.
+
+* **Convert** is a transformation that takes one or more data models
+  as input and return one or more data models as output. It is implemented
+  as a Python function.  Conversions are similar to property mappings,
+  but operates on instances instead of individual properties.
+  The *TEM image analyser* is implemented as a converter.  It takes
+  an instance of a [TEMImage] data model and returns an instance of
+  a [PrecipitateStatistics] data model.
+
+
+Setup
+-----
+Create a new virtual environment and install needed packages
+
+    pip install -r requirements.txt
+
+
+Running the example
+-------------------
+The main example can be found in the [demo.py] script. We will here go
+through it step by step.
+
+### Initialisation
+In addition to define the input and output data directories to use in this
+example, the line
+
+```python
+from paths import indir, outdir
+```
+
+also sets up search paths for DLite and include this example directory in
+the Python module search path (see [paths.py]):
+
+```python
+dlite.storage_path.append(entitydir)
+dlite.python_storage_plugin_path.append(plugindir)
+
+sys.path.append(str(exampledir))
```

Comment:
Why do you need to add this to path?

Reply:
Exactly. I guess these utilities eventually should find their way into one of our libraries?